addons/project/tests/test_project_flow.py

```diff
@@ -406,3 +406,21 @@ def test_do_not_copy_project_stage(self):
         self.project_pigs.stage_id = stage.id
         project_copy = self.project_pigs.with_context(default_stage_id=stage.id).copy()
         self.assertNotEqual(project_copy.stage_id, self.project_pigs.stage_id, 'Copied project should have lowest sequence stage')
+
+    def test_project_task_copy_without_archive_user(self):
+        self.user_projectuser.action_archive()
+        task = self.task_1.copy()
+        self.assertFalse(task.user_ids)
+        task_b = self.task_2.copy({
+            'name': 'Task B',
+            'user_ids': [Command.set([self.user_projectuser.id, self.user_projectmanager.id])],
+        })
+        self.assertEqual([self.user_projectmanager.id], task_b.user_ids.ids)
```

Comment:
To me if the user would like to set those users (by setting the users in the parameter of copy method) then we should not change the user choice.